spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java

```diff
@@ -322,33 +322,28 @@ private void updateClusterId(Admin adminClient) throws InterruptedException, Exe
 	 */
 	protected Collection<NewTopic> newTopics() {
 		Assert.state(this.applicationContext != null, "'applicationContext' cannot be null");
-		Map<String, NewTopic> newTopicsMap = new HashMap<>(
-				this.applicationContext.getBeansOfType(NewTopic.class, false, false));
-		Map<String, NewTopics> wrappers = this.applicationContext.getBeansOfType(NewTopics.class, false, false);
-		AtomicInteger count = new AtomicInteger();
-		wrappers.forEach((name, newTopics) -> {
-			newTopics.getNewTopics().forEach(nt -> newTopicsMap.put(name + "#" + count.getAndIncrement(), nt));
-		});
-		Map<String, NewTopic> topicsForRetry = newTopicsMap.entrySet().stream()
-				.filter(entry -> entry.getValue() instanceof TopicForRetryable)
-				.collect(Collectors.toMap(Entry::getKey, Entry::getValue));
-		Map<String, String> topicNameToMapKey = new HashMap<>();
-		for (Entry<String, NewTopic> entry : newTopicsMap.entrySet()) {
-			topicNameToMapKey.put(entry.getValue().name(), entry.getKey());
-		}
 
-		for (Entry<String, NewTopic> entry : topicsForRetry.entrySet()) {
-			String retryTopicName = entry.getValue().name();
-			String keyInNewTopicsMap = topicNameToMapKey.get(retryTopicName);
-			if (keyInNewTopicsMap != null) {
-				NewTopic existing = newTopicsMap.get(keyInNewTopicsMap);
-				if (!(existing instanceof TopicForRetryable)) {
-					newTopicsMap.remove(keyInNewTopicsMap);
-				}
-			}
-		}
-		newTopicsMap.entrySet().removeIf(entry -> !this.createOrModifyTopic.test(entry.getValue()));
-		return new ArrayList<>(newTopicsMap.values());
+		// Deal with List<NewTopic> directly instead of Map (no need for bean names)
+		List<NewTopic> newTopicsList = new ArrayList<>(
+				this.applicationContext.getBeansOfType(NewTopic.class, false, false).values());
+
+		// Add topics from NewTopics wrappers (no need for bean names either)
+		this.applicationContext.getBeansOfType(NewTopics.class, false, false).values()
+				.forEach(wrapper -> newTopicsList.addAll(wrapper.getNewTopics()));
+
+		// Collect normal topic names to check against TopicForRetryable
+		Set<String> normalNames = newTopicsList.stream()
```

Comment:
Just nit-pick: I don't think "normal" is a proper term to use.
And not only here.
It raises a question like what is the "abnormal"?
Is a `TopicForRetryable` abnormal?

I would suggest to go with a "regular" instead.
This way a `TopicForRetryable` is really "unregular" and that does not cause extra questions.